### PR TITLE
e2e-tests: Add filtered indexer e2e tests

### PR DESCRIPTION
**Pull request by Eric-Warehime**
## Summary

Adds an e2e test which uses a conduit writer pipeline w/ a filter processor.

Uses docker-compose w/ a conduit writer, an e2e reader w/ queries, and a pgsql db.

Followup steps: working on separating these tests into conduit and indexer.

## Test Plan

`make e2e-filter-test-nightly`

### CODE REVIEW

File: e2e_tests/docker/indexer-filtered/e2e-write/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# Misc dependencies
+ENV HOME /opt
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update && apt-get install -y apt-utils curl git git-core bsdmainutils python3 python3-pip make bash libtool libboost-math-dev libffi-dev wget gettext-base
+
+# Setup files for test
+RUN mkdir -p /opt/go/indexer
+RUN mkdir -p /opt/conduit-dir
+RUN mkdir -p /opt/algod-dir
+COPY . /opt/go/indexer
```
Comment (shiqizng) [inline e2e_tests/docker/indexer-filtered/e2e-write/Dockerfile:17]: we'll be removing conduit from this repo? should we download the conduit repo instead?

File: e2e_tests/src/e2e_indexer/e2elive.py [hunk 2]
```diff
@@ -59,30 +59,111 @@ def main():
...
-        sourcenet = e2edata and os.path.join(e2edata, "net")
-    if sourcenet and hassuffix(sourcenet, ".tar", ".tar.gz", ".tar.bz2", ".tar.xz"):
-        source_is_tar = True
     tempdir = tempfile.mkdtemp()
     if not args.keep_temps:
         atexit.register(shutil.rmtree, tempdir, onerror=logger.error)
     else:
         logger.info("leaving temp dir %r", tempdir)
+    # Sane default for lastblock--used for read_only runs
+    lastblock = 93
```
[the remark this answers is not in the document]
Comment (Eric-Warehime) [reply e2e_tests/src/e2e_indexer/e2elive.py:81]: The e2e test data currently contains 93 rounds. Usually we get this via reading the e2e data to see how many rounds we should expect to process. But in this case our read-only indexer will be in a separate container from the algod/e2e test data so we have no source of truth. 

To resolve that I've just defaulted this to 93--it still gets overwritten by the `lastblock` value in regular `e2elive.py` runs.

File: e2e_tests/docker/indexer-filtered/e2e-write/Dockerfile (hunk 1 above)
Comment (Eric-Warehime) [reply e2e_tests/docker/indexer-filtered/e2e-write/Dockerfile:17]: Yes, but when I wrote this the conduit repo didn't exist/didn't have a buildable binary.

We can either wait until everything is ready (which maybe it is now?) and update this or just do this in a followup commit.

I think I prefer following up on this in the future. For example, this PR https://github.com/algorand/indexer/pull/1505 would start failing and would force me to make this change.

File: e2e_tests/src/e2e_indexer/e2elive.py [hunk 3]
```diff
@@ -59,30 +59,111 @@ def main():
...
+    algoddir = None
+    if not args.read_only:
+        tempnet, lastblock = setup_algod(tempdir, args.source_net, args.s3_source_net)
+        algoddir = os.path.join(tempnet, "Node")
```
[the remark this answers is not in the document]
Comment (Eric-Warehime) [reply e2e_tests/src/e2e_indexer/e2elive.py]: Done!

**Review by winder — APPROVED** [review]

**Pull request merged**